### PR TITLE
Activate HANA tests in Github Actions

### PR DESCRIPTION
**Pull request by mrylov**

### CODE REVIEW

**Comment by 3nids** [comment]
please wait before #41416 is merged.

File: .docker/docker-variables.env [hunk 1]
```diff
@@ -20,3 +20,10 @@ LC_ALL=en_US.UTF-8
 PUSH_TO_CDASH=false
 
 XDG_RUNTIME_DIR=/tmp
+
+# HANA provider settings
+HANA_TESTS=true
```
Comment (3nids) [inline .docker/docker-variables.env:25]:
```suggestion
HANA_TESTS_ENABLED=ON
```

File: .docker/docker-variables.env [hunk 2]
```diff
@@ -20,3 +20,10 @@ LC_ALL=en_US.UTF-8
...
+HANA_HOST=917df316-4e01-4a10-be54-eac1b6ab15fb.hana.prod-us10.hanacloud.ondemand.com
+HANA_PORT=443
+HANA_USER=QGIS_CI
+HANA_PASSWORD="3w8dkX:NDrs&"
```
Comment (3nids) [inline .docker/docker-variables.env:29]: these lines would be better suited in .docker/docker-qgis-test.sh, no?

Comment (mrylov) [reply .docker/docker-variables.env]: I agree, it is done in the new change.

File: .docker/docker-qgis-test.sh [hunk 3]
```diff
@@ -10,9 +10,13 @@ set -e
 ##################################
 
 if [ ${HANA_TESTS_ENABLED:-"OFF"} == "ON" ] ; then
-  echo "::group::hana"
```
Comment (3nids) [inline .docker/docker-qgis-test.sh:13]: this is used to have nicer output on the workflow.
any reason to remove this?

Comment (mrylov) [reply .docker/docker-qgis-test.sh:13]: I did it by analogue with other database initialization sections (PostgreSQL, Oracle) where it was removed.

Comment (3nids) [reply .docker/docker-qgis-test.sh:13]: Can you replace them? They are useful in the log.
Otherwise LGTM

Comment (mrylov) [reply .docker/docker-qgis-test.sh:13]: Sure, done.

**Comment by nyalldawson** [comment]
@mrylov the CI is now reporting failures in the HANA connection test -- see e.g. https://github.com/qgis/QGIS/runs/1992557118 . Can you take a look and see what's happening here?

**Comment by mrylov** [comment]
@nyalldawson, sure, I will take a look.

**Comment by mrylov** [comment]
@nyalldawson, I've double checked the HANA provider and found no issues.
Interestingly, the [latest builds](https://github.com/qgis/QGIS/runs/2019627711?check_suite_focus=true) do not report anything in the HANA provider.

**Comment by stefanuhrig** [comment]
@mrylov There seem to be sporadic failures of the HANA test on the CI (https://github.com/qgis/QGIS/pull/42081#issuecomment-792463728). I think we need to have a look what's causing that.

**Comment by mrylov** [comment]
@stefanuhrig, I've started looking into the issue.

**Comment by mrylov** [comment]
@nyalldawson, the issue with failed HANA tests has been fixes in #42098.

**Pull request merged**